### PR TITLE
Escape single quote if execute in clean env

### PR DESCRIPTION
**Pull request by juztas**

### CODE REVIEW

**Comment by juztas** [comment]
https://github.com/dmwm/CRABClient/pull/4594

**Comment by juztas** [comment]
@mmascher @jmarra13 this should solve for users with older release. I tried with 5_3_19, 7_0_6, 7_6_0 and it works ok.

**Comment by mmascher** [comment]
@juztas can you create the "1.0.12_crab version of" this pull request please (like https://github.com/dmwm/WMCore/pull/6478)

Then we will need a WMCore tag if we want to deploy this quickly (depending if the user still has problems)

**Comment by juztas** [comment]
@mmascher https://github.com/dmwm/WMCore/pull/6571

File: src/python/WMCore/Credential/Proxy.py [hunk 1]
```diff
@@ -159,6 +159,8 @@ def setEnv(self, cmd):
         if self.uisource is not None and len(self.uisource) > 0:
             ret = 'source ' + self.uisource + ' && ' + ret
         if self.cleanEnvironment:
+            # Need to escape ' in cmd
+            ret = ret.replace("'", "'\\''")
```
Comment (ticoann) [inline src/python/WMCore/Credential/Proxy.py]: isn't this following?
```python
ret = ret.replace("'", "\\'")
or 
ret = ret.replace("'", r"\'")
```

**Comment by juztas** [comment]
updated in both pull requests to use r

File: src/python/WMCore/Credential/Proxy.py [hunk 2]
```diff
@@ -159,6 +159,8 @@ def setEnv(self, cmd):
         if self.uisource is not None and len(self.uisource) > 0:
             ret = 'source ' + self.uisource + ' && ' + ret
         if self.cleanEnvironment:
+            # Need to escape ' in cmd
+            ret = ret.replace("'", r"'\''")
```
Comment (ticoann) [inline src/python/WMCore/Credential/Proxy.py]: Hi Justas, 

I meant whether there need additional sing quote?
``` python
ret = ret.replace("'", r"'\''") # this produces "Tom'\\''s house"
ret = ret.replace("'", r"\'") # this produces "Tom\\'s house"
```
I was just wondering the first one is what you want (I thought it should be the second case)
first one produces

**Pull request merged**